src/lib/node.ts

```diff
@@ -230,13 +231,25 @@ export function pipeToNodeStream<T>(
  * @param nodeReadable Soource NodeJS stream
  * @returns Readable stream of type `string | Buffer`
  */
-export function fromNodeReadable(
+export function fromNodeReadable<T = void>(
 	nodeReadable: NodeJS.ReadableStream
-): ReadableStream<string | Buffer> {
-	const stream = new Stream<string | Buffer>();
+): ReadableStream<FromNodeReadableDataType<T>> {
+	const stream = new Stream<FromNodeReadableDataType<T>>();
 
 	const endStream = async (err?: Error) => {
-		if (!stream.isEndingOrEnded()) return stream.end(err);
+		if (!stream.isEndingOrEnded()) {
+			/**
+			 * `stream.end` could reject, so we'll emit
+			 * any caught errors as an event via the NodeJS stream.
+			 * This is acceptable as we are only attempting to end
+			 * the ts-stream a maximum one time.
+			 */
+			try {
+				return stream.end(err);
+			} catch (e) {
+				nodeReadable.emit("error", e);
```

Comment:
I believe events should only ever be emitted by an object itself, not by any external entity (i.e. I think `emit()` should actually have been a protected method).
To ensure correct internal state of the object (nodeReadable), I prefer using a public API if it's available, and I think that should be the [`destroy()`](https://nodejs.org/api/stream.html#stream_readable_destroy_error) method in this case, which also accepts an optional error.